File: experiments/03_synth_data_generation/output/temp_batch02_validation.c
#include <assert.h>

#include "/Users/brandomiranda/lean-ebm/experiments/03_synth_data_generation/c_src/batch02_gcd_math.c"

int main(void) {
    assert(gcd(0u, 0u) == 0u);
    assert(gcd(1u, 0u) == 1u);
    assert(gcd(48u, 18u) == 6u);

    assert(lcm(0u, 5u) == 0u);
    assert(lcm(1u, 7u) == 7u);
    assert(lcm(12u, 18u) == 36u);

    assert(power(0u, 0u) == 1u);
    assert(power(5u, 0u) == 1u);
    assert(power(2u, 10u) == 1024u);

    assert(is_prime(0u) == 0);
    assert(is_prime(2u) == 1);
    assert(is_prime(97u) == 1);
    assert(is_prime(99u) == 0);

    assert(mod_exp(7u, 3u, 0u) == 0u);
    assert(mod_exp(5u, 0u, 13u) == 1u);
    assert(mod_exp(2u, 10u, 1000u) == 24u);

    assert(sum_digits(0u) == 0u);
    assert(sum_digits(7u) == 7u);
    assert(sum_digits(12345u) == 15u);

    assert(count_digits(0u) == 1u);
    assert(count_digits(7u) == 1u);
    assert(count_digits(1000u) == 4u);

    return 0;
}

File: experiments/03_synth_data_generation/output/temp_batch02_validation.py
import importlib.util


spec = importlib.util.spec_from_file_location(
    "batch02_gcd_math",
    "/Users/brandomiranda/lean-ebm/experiments/03_synth_data_generation/py_src/batch02_gcd_math.py",
)
module = importlib.util.module_from_spec(spec)
assert spec.loader is not None
spec.loader.exec_module(module)

assert module.gcd(0, 0) == 0
assert module.gcd(1, 0) == 1
assert module.gcd(48, 18) == 6

assert module.lcm(0, 5) == 0
assert module.lcm(1, 7) == 7
assert module.lcm(12, 18) == 36

assert module.power(0, 0) == 1
assert module.power(5, 0) == 1
assert module.power(2, 10) == 1024

assert module.is_prime(0) is False
assert module.is_prime(2) is True
assert module.is_prime(97) is True
assert module.is_prime(99) is False

assert module.mod_exp(7, 3, 0) == 0
assert module.mod_exp(5, 0, 13) == 1
assert module.mod_exp(2, 10, 1000) == 24

assert module.sum_digits(0) == 0
assert module.sum_digits(7) == 7
assert module.sum_digits(12345) == 15

assert module.count_digits(0) == 1
assert module.count_digits(7) == 1
assert module.count_digits(1000) == 4
